### ISSUE TITLE
ingress not routing

### ISSUE THREAD

**Issue by dudo** [kubernetes-sigs/aws-alb-ingress-controller#556]
I seem to have something configured poorly... I had this working nicely with the nginx-ingress tooling, but something seems different with this alb adaptation.

```
$ kubectl get ing --all-namespaces
NAMESPACE                    NAME                HOSTS                                                  ADDRESS                                                                 PORTS     AGE
echoserver                   echoserver          *.nonprod.smartbizloans.com                            4d480b86-echoserver-echose-2ad7-318103396.us-west-2.elb.amazonaws.com   80        4h
veritax-api-rewrite-docker   veritax-api-rewrite-docker   testing.nonprod.smartbizloans.com                                                                           80        2m
```

Before, there was an a url attached to a service, and i added that to the cname for *.nonprod.smartbizloans.com. Upon adding ingress controllers with the appropriate host (testing.nonprod...), everything was picked up, and life was good.

With this tooling, things are a bit different. There is the echoserver namespace, which has a service with no external ip. The external IP is set up on another ingress (which is confusing me). I feel like I set the wrong host on the "parent" ingress, but I'm not sure how to proceed. 

http://veritax-api-rewrite-docker.nonprod.smartbizloans.com/

The ALB seems to be responding, but the "child" ingresses aren't being respected.

**Comment by dudo** [comment]
@bigkraig Can you point me in the right direction here?

**Comment by bigkraig** [comment]
I don't fully understand what you are trying to set up. FWIW, the ingress controller should only add the ALB hostname (which external-dns uses) to the ingress resource. Services wouldn't have anything from the controller. A service might get an external name if its a `servicetype=loadbalancer`.

Do you have multiple layers of ingress because some feature is missing from ALBs? I see rewrite in there..

**Comment by dudo** [comment]
By layers, I mean that I have a handful of apps that get routed according to their ingress.

`/` goes to marketing
`/apply` goes to js application
`/admin` goes to core backend app

Each "service" sets up their own ingress. When I first started tinkering with this, I used nginx-ingress, and that worked great. There was a kubernetes service that exposed an endpoint, and it listened to the various ingress that I applied. I added that endpoint to a cname record for *.nonprod.smartbizloans.com DNS and it found it's way to the appropriate ingress that I defined.

With this implementation, there is no service - just this echoserver ingress. I guess my question is, what is this echo server supposed to represent? It *is* the ALB, right? And is the host supposed to be echoserver.smartbizloans.com (in the ingress manifest)? And where do I point my cname entries to? 

echoserver.smartbizloans.com -> ?
*.nonprod.smartbizloans.com -> ?

Is this making sense? Wildcard be damned, how are you supposed to set this up to have ingress records respected?
